oscarpilote/Ortho4XP
# Alternative installation for Windows 10

# Background
Currently, the windows installation requires you to install Ortho4XP python dependencies in global pip package system. The main drawbacks of this method are that your other python project dependencies gets polluted and hence often won't work at all. In short, installing Ortho4XP python dependencies will cause other python projects not to work because of package version conflicts.

# Solution
Pip is a package management tool for python and Ortho4XP uses pip to install. But if we keep installing packages to its global repository, then we might someday run into issues like conflicting packages with other python projects. For example, Let's say another X-Plane based python project uses the same package that Ortho4XP uses, but a different version. How are we going to install that project without causing Ortho4XP not to function? The solution is to separate every projects' dependencies. This is exactly what I'm trying to do for Ortho4XP. We can install Ortho4XP as a standalone app in the user's computer and never have to worry about conflicting packages with other projects. We definitely don't want to pollute users global python packages and cause other python projects not to work or vice versa.

# How it works
There are a couple of different ways of Installing python packages into the user's computer. Using pip is one of the popular options. Pip has a package called virtualenv which allows us to create a virtual python environment for each project and set up its packages on its own - completely separating it from global `pip` space. This is a great advantage that we are going to utilize in this PR. This ensures Ortho4XP to be a true standalone app with its own dependencies installed.

# Steps to test in Windows 10:
1. Follow Step 1 for windows installation in the original `Install_Instructions.txt`. That is to install Python, pip and make sure to add python to your environment PATH variable.
2. Open up the PowerShell command prompt using Administrator privilege. (optional unless you prefer git clone in step 3).
3. Git clone the project/unzip the Ortho4XP project and cd/browse into the project directory.
4. Run `Windows10_Setup.bat`. This should take care of all the necessary installation.

# Running the Ortho4XP app
1. Run `Windows10_Ortho4XP.bat` by double-clicking or using PowerShell command prompt. This will activate the virtual environment for you and run `Ortho4XP_v130.py`.

# Requirements
1. Windows 10 (This is a test OS, for now if we get enough support, we can think of supporting other OS as well)
2. Python 3.6 / Python 3.7 (I can work on supporting other versions)
3. Follow Step 1 for windows installation in the original `Install_Instructions.txt`.

@oscarpilote: Please review.
